src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,2233 @@
+  SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
+                     "ARKODE::lsrkStep_TakeStepSSP104", "stage RHS",
+                     "F(:) =", "");
+  N_VPrintFile(ark_mem->fn, ARK_LOGGER->debug_fp);
+#endif
+
+  N_VScale(ONE, ark_mem->yn, ark_mem->tempv2);
+
+  N_VLinearSum(ONE, ark_mem->yn, onesixth * ark_mem->h, ark_mem->fn,
+               ark_mem->ycur);
+  if (!ark_mem->fixedstep)
+  {
+    N_VLinearSum(ONE, ark_mem->yn, onefifth * ark_mem->h, ark_mem->fn,
+                 ark_mem->tempv1);
+  }
+
+  /* Evaluate stages j = 2,...,step_mem->req_stages */
+  for (int j = 2; j <= 5; j++)
+  {
+#if SUNDIALS_LOGGING_LEVEL >= SUNDIALS_LOGGING_DEBUG
+    SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
+                       "ARKODE::lsrkStep_TakeStepSSP104", "start-stage",
+                       "step = %li, stage = %i, h = %" RSYM ", tcur = %" RSYM,
+                       ark_mem->nst, j, ark_mem->h,
+                       ark_mem->tcur +
+                         ((sunrealtype)j - ONE) * onesixth * ark_mem->h);
+#endif
+
+    /* apply user-supplied stage postprocessing function (if supplied) */
+    if (ark_mem->ProcessStage != NULL)
+    {
+      retval = ark_mem->ProcessStage(ark_mem->tcur + ((sunrealtype)j - TWO) *
+                                                       onesixth * ark_mem->h,
+                                     ark_mem->ycur, ark_mem->user_data);
+      if (retval != 0) { return ARK_POSTPROCESS_STAGE_FAIL; }
+    }
+
+    retval = step_mem->fe(ark_mem->tcur +
+                            ((sunrealtype)j - ONE) * onesixth * ark_mem->h,
+                          ark_mem->ycur, ark_mem->fn, ark_mem->user_data);
+    step_mem->nfe++;
+    if (retval < 0) { return ARK_RHSFUNC_FAIL; }
+    if (retval > 0) { return RHSFUNC_RECVR; }
+
+#ifdef SUNDIALS_LOGGING_EXTRA_DEBUG
+    SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
+                       "ARKODE::lsrkStep_TakeStepSSP104", "stage RHS",
+                       "F_%i(:) =", j);
+    N_VPrintFile(ark_mem->fn, ARK_LOGGER->debug_fp);
+#endif
+
+    N_VLinearSum(ONE, ark_mem->ycur, onesixth * ark_mem->h, ark_mem->fn,
+                 ark_mem->ycur);
+    if (j == 4 && !ark_mem->fixedstep)
+    {
+      N_VLinearSum(ONE, ark_mem->tempv1, SUN_RCONST(0.3) * ark_mem->h,
+                   ark_mem->fn, ark_mem->tempv1);
+    }
+  }
+  N_VLinearSum(SUN_RCONST(1.0) / SUN_RCONST(25.0), ark_mem->tempv2,
+               SUN_RCONST(9.0) / SUN_RCONST(25.0), ark_mem->ycur,
+               ark_mem->tempv2);
+  N_VLinearSum(SUN_RCONST(15.0), ark_mem->tempv2, SUN_RCONST(-5.0),
+               ark_mem->ycur, ark_mem->ycur);
+
+  /* apply user-supplied stage postprocessing function (if supplied) */
+  if (ark_mem->ProcessStage != NULL)
+  {
+    retval = ark_mem->ProcessStage(ark_mem->tcur + FOUR * onesixth * ark_mem->h,
+                                   ark_mem->ycur, ark_mem->user_data);
+    if (retval != 0) { return ARK_POSTPROCESS_STAGE_FAIL; }
+  }
```

Comment:
See suggestion below moving this inside the loop
```suggestion
```

Reply:
I cannot move this inside since the pattern changes immediately after this abscissa. We have `c = 1/6*(0, 1, 2, 3, 4, 2, 3, 4, 5, 6)`. I believe we discussed this in the lab and decided to keep it in this form. Is there anything wrong with the current setup?

Reply:
I think this is currently evaluated at the wrong stage time and by moving it inside the loop and updating the stage time so it matches the time used in the RHS evaluation in the loop (see the suggested addition below) fixes this.

Reply:
I had initially thought that the ProcessStage time needed to match the time of the last RHS call. However, if it should actually match the time of the upcoming RHS call, as I inferred from your corrections, then slight changes as in the commit [a471db6](https://github.com/LLNL/sundials/pull/541/commits/a471db643703ecedd9f563fae7c8c2c896bec3d3) should resolve the inconsistency in timing.

Reply:
Yes, the time passed to `ProcessStage` should correspond to the time that stage represents which is also the time passed to the following RHS call with that solution value.